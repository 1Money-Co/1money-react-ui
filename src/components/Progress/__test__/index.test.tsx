import 'jsdom-global/register';
import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Progress } from '../index';

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Progress', () => {
  it('renders correctly', () => {
    const ring = render(
      <Progress />
    );
    const bar = render(
      <Progress type="bar" />
    );
    expect(ring).toMatchSnapshot();
    expect(bar).toMatchSnapshot();
  });
});
