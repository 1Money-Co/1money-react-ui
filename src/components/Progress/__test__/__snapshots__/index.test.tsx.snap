// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="usd1-react-ui-progress"
      >
        <div
          class="usd1-react-ui-progress-ring"
          style="width: 106px; height: 106px; --progress: 0%; --last: calc(100% - 0%); --size: 106px; --stroke: 6px; --bg-color: #FAFAFC; --trail-color: #E0E2EE; --fill-color: #3D73F2;"
        />
      </div>
    </div>
    <div>
      <div
        class="usd1-react-ui-progress"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="usd1-react-ui-progress-bar p-progressbar p-component p-progressbar-determinate"
          data-pc-name="progressbar"
          data-pc-section="root"
          role="progressbar"
          style="--trail-color: #E0E2EE; --fill-color: #3D73F2; --label-color: #FFFFFF;"
        >
          <div
            class="p-progressbar-value p-progressbar-value-animate"
            data-pc-section="value"
            style="width: 0%; display: flex;"
          >
            <div
              class="p-progressbar-label"
              data-pc-section="label"
            >
              0%
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="usd1-react-ui-progress"
    >
      <div
        class="usd1-react-ui-progress-ring"
        style="width: 106px; height: 106px; --progress: 0%; --last: calc(100% - 0%); --size: 106px; --stroke: 6px; --bg-color: #FAFAFC; --trail-color: #E0E2EE; --fill-color: #3D73F2;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Progress renders correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="usd1-react-ui-progress"
      >
        <div
          class="usd1-react-ui-progress-ring"
          style="width: 106px; height: 106px; --progress: 0%; --last: calc(100% - 0%); --size: 106px; --stroke: 6px; --bg-color: #FAFAFC; --trail-color: #E0E2EE; --fill-color: #3D73F2;"
        />
      </div>
    </div>
    <div>
      <div
        class="usd1-react-ui-progress"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="usd1-react-ui-progress-bar p-progressbar p-component p-progressbar-determinate"
          data-pc-name="progressbar"
          data-pc-section="root"
          role="progressbar"
          style="--trail-color: #E0E2EE; --fill-color: #3D73F2; --label-color: #FFFFFF;"
        >
          <div
            class="p-progressbar-value p-progressbar-value-animate"
            data-pc-section="value"
            style="width: 0%; display: flex;"
          >
            <div
              class="p-progressbar-label"
              data-pc-section="label"
            >
              0%
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="usd1-react-ui-progress"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="usd1-react-ui-progress-bar p-progressbar p-component p-progressbar-determinate"
        data-pc-name="progressbar"
        data-pc-section="root"
        role="progressbar"
        style="--trail-color: #E0E2EE; --fill-color: #3D73F2; --label-color: #FFFFFF;"
      >
        <div
          class="p-progressbar-value p-progressbar-value-animate"
          data-pc-section="value"
          style="width: 0%; display: flex;"
        >
          <div
            class="p-progressbar-label"
            data-pc-section="label"
          >
            0%
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
